src/gss_names.c

```
@@ -626,9 +704,108 @@ uint32_t gssntlm_inquire_name(uint32_t *minor_status,
+        buf.length = attr_value->length + attr_name_len + 2;
+        buf.value = malloc(buf.length);
+        if (buf.value == NULL) {
+            retmin = ENOMEM;
+            goto done;
+        }
+
+        /* Construct 'attr_name=<attr_value>\0' string */
+        memcpy(buf.value, attr->attr_name, attr_name_len);
+        ((char*)buf.value)[attr_name_len] = '=';
+        memcpy(((char*)buf.value) + attr_name_len + 1, attr_value->value, attr_value->length);
+        ((char*)buf.value)[buf.length] = 0;
+
+        retmaj = gss_add_buffer_set_member(&retmin, &buf, attrs);
+        free(buf.value);
+        if (retmaj != GSS_S_COMPLETE) goto done;
+
+        i++;
+    }
+    return GSSERRS(0, GSS_S_COMPLETE);
+
+done:
+    (void)gss_release_buffer_set(&tmpmin, attrs);
+    return GSSERRS(retmin, retmaj);
+}
+
+/* RFC6680 - GSSAPI Naming Extensions */
+uint32_t gssntlm_get_name_attribute(uint32_t *minor_status,
+                                    gss_name_t name,
+                                    gss_buffer_t attr,
+                                    int *authenticated,
+                                    int *complete,
+                                    gss_buffer_t value,
+                                    gss_buffer_t display_value,
+                                    int *more)
+{
+    uint32_t retmin;
+    uint32_t retmaj;
+
+    int index;
+    const struct gssntlm_name *in = (const struct gssntlm_name *)name;
+
+    if (name == GSS_C_NO_NAME) {
+        return GSSERRS(GSS_S_BAD_NAME, GSS_S_CALL_INACCESSIBLE_READ);
+    }
+    if (attr == NULL) {
+        return GSSERRS(ERR_NOARG, GSS_S_CALL_INACCESSIBLE_READ);
+    }
+
+    if (display_value != NULL) {
+        display_value->value = NULL;
+        display_value->length = 0;
+    }
+    if (more != NULL) *more = 0;
+    index = gssntlm_find_attr(in->attrs, attr->value, attr->length);
+    if (index == -1) {
+        if (authenticated) *authenticated = 0;
```

We should probably zero these at function start so we do not have to care to do it everywhere we might want to return an error in future.

---

np